digitalocean/certificate/datasource_certificate.go

```diff
@@ -93,37 +93,19 @@ func dataSourceDigitalOceanCertificateRead(ctx context.Context, d *schema.Resour
 }
 
 func FindCertificateByName(client *godo.Client, name string) (*godo.Certificate, error) {
-	opts := &godo.ListOptions{
-		Page:    1,
-		PerPage: 200,
-	}
-
-	for {
-		certs, resp, err := client.Certificates.List(context.Background(), opts)
-		if resp != nil && resp.StatusCode == http.StatusNotFound {
-			return nil, nil
-		}
-		if err != nil {
-			return nil, fmt.Errorf("Error retrieving certificates: %s", err)
-		}
-
-		for _, cert := range certs {
-			if cert.Name == name {
-				return &cert, nil
-			}
-		}
 
-		if resp.Links == nil || resp.Links.IsLastPage() {
-			break
-		}
+	cert, resp, err := client.Certificates.ListByName(context.Background(), name, nil)
+	if resp != nil && resp.StatusCode == http.StatusNotFound {
+		return nil, nil
+	}
 
-		page, err := resp.Links.CurrentPage()
-		if err != nil {
-			return nil, fmt.Errorf("Error retrieving certificates: %s", err)
-		}
+	if err != nil {
+		return nil, fmt.Errorf("Error retrieving certificates: %s", err)
+	}
 
-		opts.Page = page + 1
+	if len(cert) == 0 {
+		return nil, fmt.Errorf("certificate not found")
 	}
 
-	return nil, fmt.Errorf("Certificate %s not found", name)
+	return &cert[0], err
```

Comment:
is there a constraint somewhere that name has to be unique? what if there are certificates with the same name and multiple certs are returned?

Reply:
name is going to be unique we already have a unique index in db. infact development of ticket started with the fact that name is unique and we should make it convenient for users to get certs by name